compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -418,9 +420,40 @@ public void visit(BLangRecordTypeNode recordTypeNode) {
             rewrite(field, recordTypeNode.typeDefEnv);
         }
         recordTypeNode.restFieldType = rewrite(recordTypeNode.restFieldType, env);
+        generateClosuresForDefaultValuesInTypeRefs(recordTypeNode, typeSymbol);
         result = recordTypeNode;
     }
 
+    private void generateClosuresForDefaultValuesInTypeRefs(BLangRecordTypeNode recordTypeNode,
```

Comment:
Let's use proper terms for additions - inclusions instead of refs. Also use parameter names that better reflect what the param is.